tests/autoblocks/test_run_test_suite.py

```diff
@@ -0,0 +1,693 @@
+import dataclasses
+import os
+from unittest import mock
+
+import pytest
+
+from autoblocks._impl.testing.models import BaseTestCase
+from autoblocks._impl.util import AutoblocksEnvVar
+from autoblocks.testing.models import BaseEvaluator
+from autoblocks.testing.models import Evaluation
+from autoblocks.testing.models import Threshold
+from autoblocks.testing.run import run_test_suite
+from tests.autoblocks.util import decode_request_body
+from tests.autoblocks.util import make_expected_body
+
+CLI_SERVER_ADDRESS = "http://localhost:8080"
+
+
+@pytest.fixture(autouse=True)
+def mock_cli_server_address_env_var():
+    with mock.patch.dict(
+        os.environ,
+        {
+            AutoblocksEnvVar.CLI_SERVER_ADDRESS.value: CLI_SERVER_ADDRESS,
+        },
+    ):
+        yield
+
+
+@dataclasses.dataclass()
+class MyTestCase(BaseTestCase):
+    input: str
+
+    def hash(self) -> str:
+        return self.input
+
+
+def test_no_test_cases(httpx_mock):
+    httpx_mock.add_response()
+
+    run_test_suite(
+        id="my-test-id",
+        test_cases=[],
+        evaluators=[],
+        fn=lambda _: None,
+        max_test_case_concurrency=1,
+        max_evaluator_concurrency=1,
```

Comment:
is it easy to add a test to make sure the concurrency params are respected?

Reply:
i think what i can do is just add a test that asserts on the order of the requests made, since it should be deterministic if both concurrency settings are set to 1